File: android/src/main/java/com/azifacemobile/theme/ReadyScreen.java
package com.azifacemobile.theme;

import android.graphics.Typeface;

import com.azifacemobile.theme.abstracts.ViewStyle;
import com.azifacemobile.utils.Theme;
import com.facebook.react.bridge.ReactApplicationContext;

import org.json.JSONObject;

import javax.annotation.Nullable;

public class ReadyScreen extends ViewStyle {
  private static final String KEY = "readyScreen";
  private final JSONObject target;
  private final Font font;
  private final Color color;

  public ReadyScreen(ReactApplicationContext context, JSONObject target) {
    super(target, KEY);

    this.font = new Font(context);
    this.color = new Color();

    this.target = new Theme().getTarget(target, KEY);
  }

  @Override
  public int getTextBackgroundColor() {
    return super.getTextBackgroundColor(android.graphics.Color.TRANSPARENT);
  }

  public int getHeaderTextColor() {
    return this.color.getColor(this.target, "headerTextColor", "#272937");
  }

  @Nullable
  public Typeface getHeaderFont() {
    return this.font.getTypography(this.target, "headerFont");
  }

  public int getSubtextColor() {
    return this.color.getColor(this.target, "subtextColor", "#272937");
  }

  @Nullable
  public Typeface getSubtextFont() {
    return this.font.getTypography(this.target, "subtextFont");
  }

  public int getOvalFillColor() {
    return this.color.getColor(this.target, "ovalFillColor", android.graphics.Color.TRANSPARENT);
  }
}

File: ios/Theme/ReadyScreen.swift
public class ReadyScreen {
  private static let TRANSPARENT: String = "#00000000"
  private let style: Style
  private let target: NSDictionary?
  private let font: Font
  private let color: Color

  init(target: NSDictionary?) {
    self.style = Style()
    self.font = Font()
    self.color = Color()

    self.target = self.style.getTarget(target, key: "readyScreen")
  }

  public func getHeaderTextColor() -> UIColor {
    return self.color.getColor(self.target, key: "headerTextColor", defaultColor: "#000000")
  }
  
  public func getHeaderFont() -> UIFont {
    return self.font.getTypography(theme: self.target, key: "headerFont")
  }

  public func getSubtextColor() -> UIColor {
    return self.color.getColor(self.target, key: "subtextColor", defaultColor: "#000000")
  }
  
  public func getSubtextFont() -> UIFont {
    return self.font.getTypography(theme: self.target, key: "subtextFont")
  }

  public func getOvalFillColor() -> UIColor {
    return self.color.getColor(self.target, key: "ovalFillColor", defaultColor: ReadyScreen.TRANSPARENT)
  }
  
  public func getTextBackgroudColor() -> UIColor {
    return self.color.getColor(self.target, key: "textBackgroundColor", defaultColor: ReadyScreen.TRANSPARENT)
  }
  
  public func getTextBackgroudColorCornerRadius() -> Int32 {
    let key = "textBackgroundColorCornerRadius"
    let defaultCornerRadius: Int32 = -1
    if !self.style.exists(self.target, key: key) {
        return defaultCornerRadius
    }
      
    return (target?[key] as? Int32) ?? defaultCornerRadius
  }
}
